File: cpp/questions/maxSubArray.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxs = nums[0];
        int curs = 0;
        for (const auto n: nums) {
            if (curs < 0) {
                curs = n;
            } else {
                curs = curs + n;
            }
            maxs = std::max(maxs, curs);
        }
        return maxs;
    }
};

File: python/questions/maxSubArray.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxs = nums[0]
        curs = 0
        for n in nums:
            curs = n if curs < 0 else curs + n
            if curs > maxs:
                maxs = curs
        return maxs
